### PR TITLE
22 create home page for signed in users

### LINKED ISSUE #22: Create : Home Page for Signed In Users
## Summary

Home page after the sign in should look different. You can welcome the user with a small banner and with a call to action button that says "Can't see what you need?" which can redirect the user to AllItemsPage and "Would you like to add an item" which brings up the AddItemForm, or instead of redirecting the user, you can put the search bar directly in the banner and show itemCards with the appropriate name, then we must showcase the most viewed items in a carousel. Since there will be three main categories for the items, which are Handcrafted - Second Hand - Donation, we can show most viewed items in carousels of their own category.

Team that will be assigned to this issue should also create the carousel component that will be used inside this page. If tailwind carousels would not be efficient, there are a lot of minor libraries for react carousels.

You can create some mock data about some items, which will consist of a picture, a name, a description, a date(when it was put on the website), location, and viewcount, import that mock data and render ItemCards from that mock data, just to test the carousels and show some data until we plug in to our backend.

Design and responsiveness must be readied with Tailwind, snapshot/unit tests should also be written.

### Screenshots

As always, this is just to give some idea, dont build the same thing.

itemPage

### Basic example

N/A

### Motivation

After the sign in, we want the users to encounter some of the items website has to offer.

### PR DESCRIPTION
**Pull request by OmarBassatni97**

### CODE REVIEW

**Review by OussamaMasri — APPROVED** [review]
Great Job Guys as always
small suggestions
can we just remove the edit button from the HomeScreen because it doesn't make any sense 
and Idk why but I felt that dog is really staring at my soul :)
so if we can make it a bit smaller(the cards) that would be perfect 
other then that You guys did an Amazing Job

**Comment by OmarBassatni97** [comment]
> Great Job Guys as always small suggestions can we just remove the edit button from the HomeScreen because it doesn't make any sense and Idk why but I felt that dog is really staring at my soul :) so if we can make it a bit smaller(the cards) that would be perfect other then that You guys did an Amazing Job

Yeah of course it doesn't make sense but its not me who placed it there that's why I didn't change anything there and for the ItemCard this is just for the Demo we are still waiting for the ItemCard that will be for the whole project

File: src/Pages/SignedInUsersHomePage/SignedInUsersHomePage.test.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+// test(' SignedInUsersHomePage renders without crashing ' , () => {
+//     const tree = renderer.create(
+//         <SignedInUsersHomePage>
+//             <ItemCard />
+//         </SignedInUsersHomePage>
+//      ).toJSON();
+//     expect(tree).toMatchSnapshot();
+
+// })
```
Comment (Iseluin) [inline src/Pages/SignedInUsersHomePage/SignedInUsersHomePage.test.js]: This seems like its not doing anything there, can you delete it ?

File: src/Pages/SignedInUsersHomePage/SignedInUsersHomePage.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+                </div>
+            </div>
+            <div className='mx-4 lg:mx-0'>
+                <h1 className='text-2xl md:text-3xl font-bold mt-2 mb-2'>Some of our products</h1>
+                <div className='flex flex-col md:grid grid-cols-3 gap-6 mt-2'>
+                    <ItemCard imgSrc="https://placeimg.com/300/300/architecture" />
+                    <ItemCard imgSrc="https://placeimg.com/300/300/animals" />
+                    <ItemCard imgSrc="https://placeimg.com/300/300/architecture" />
+                </div>
+            </div>
```
Comment (Iseluin) [inline src/Pages/SignedInUsersHomePage/SignedInUsersHomePage.js]: Can we put a slider in here which showcases items in a slider for each main category that we have ?

File: src/Pages/SignedInUsersHomePage/SignedInUsersHomePage.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    return (
+        <div data-testid='SignedInUseresHomePage' className='bg-background text-primary flex flex-col items-center justify-between'>
+            <div className=' bg-home-page-for-signed-in-users bg-cover w-full min-h-[40vh] md:min-h-[7p0vh] text-background p-5'>
+                <div className=' mb-3 md:mb-0 pt-5 md:pt-14 md:pb-6 w-full'>
+                    <h1 className='text-2xl md:text-5xl font-bold text-center'>Welcome back, </h1>
+                    <h1 className='text-2xl md:text-5xl font-bold text-center'> {userData.firstName} {userData.lastName}</h1>
+                    <h3 className='text-l md:text-2xl font-semi-bold text-center'> Dilmun is your best choice to buy or sell anything</h3>
+                    <div className='flex flex-row flex-wrap gap-4 justify-center  md:text-2xl m-4'>
+                        <Link className='px-3 py-2 rounded bg-secondary ml-2 hover:text-tertiary hover:bg-primary transition duration-300' to={PRODUCT_ROUTE}>More Products</Link>
+                        <Link className='px-3 py-2 rounded bg-secondary ml-2 hover:text-tertiary hover:bg-primary transition duration-300' to={ADD_ITEM_ROUTE}>Add New Item</Link>
```
Comment (Iseluin) [inline src/Pages/SignedInUsersHomePage/SignedInUsersHomePage.js]: Can we make these buttons a bit fancier? Like they dont even need to look like a button with the blue background, we can use icons and show them as links as well.

File: src/Pages/SignedInUsersHomePage/SignedInUsersHomePage.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+import { user } from '../../features/user/userSlice'
+import { ADD_ITEM_ROUTE, PRODUCT_ROUTE } from '../../route'
+import ItemCard from './ItemCard'
+
+
+const SignedInUsersHomePage = () => {
+    const userData = useSelector(user)
+    return (
+        <div data-testid='SignedInUseresHomePage' className='bg-background text-primary flex flex-col items-center justify-between'>
+            <div className=' bg-home-page-for-signed-in-users bg-cover w-full min-h-[40vh] md:min-h-[7p0vh] text-background p-5'>
```
Comment (Iseluin) [inline src/Pages/SignedInUsersHomePage/SignedInUsersHomePage.js]: The resolution of the bg image looks low, can we use a higher resolution one ?

**Review by Iseluin — CHANGES_REQUESTED** [review]
Some minor comments, looks good overall!

**Comment by karzan7899** [comment]
have you updated the tests?

**Comment by OmarBassatni97** [comment]
> have you updated the tests?

Yes, now

**Review by sarahaboudalal — COMMENTED** [review]
Looking good guys, i just have few comments:
1) u have some typos in the hero section, and like add a "?" after the question and stuff, little minor edits basically
2) for the photo slide i suggest u add transitions and duration-300? idk try it out
other than that, it looks good

**Review by Iseluin — CHANGES_REQUESTED** [review]
Looks gorgeous @OmarBassatni97 @haboosh92 , needs just some small touches:

1. The background image, can you blur it just like what we have on the hero section of the home page?
2. Please make the "Some of our Products" into "Most Recent Items", make it bold and center it.
3. Please make the cards smaller so the user can see them without scrolling down.
4. You can remove the unnecessary components from the app like 404, alert etc.
5. You can make the background image even smaller(Heightwise).

**Review by Iseluin — APPROVED** [review]
Nice!

**Pull request merged**